stl/inc/atomic

```diff
@@ -418,21 +418,42 @@ void _Atomic_wait_direct(
 }
 #endif // _HAS_CXX20
 
-#if 1 // TRANSITION, ABI
+#if 1 // TRANSITION, ABI, GH-1151
 inline void _Atomic_lock_acquire(long& _Spinlock) noexcept {
-    while (_InterlockedExchange(&_Spinlock, 1)) {
-        _YIELD_PROCESSOR();
+#if defined(_M_IX86) || defined(_M_X64)
+    // Algorithm from Intel(R) 64 and IA-32 Architectures Optimization Reference Manual, September 2019

```

Comment:
```suggestion
    // Algorithm from Intel(R) 64 and IA-32 Architectures Optimization Reference Manual, May 2020

```

Reply:
(I checked that it's still on the same page)

Reply:
Alternately I'm not sure we need to mention this at all, I don't think there's anything novel being consumed from the optimization manual here beyond "you should implement exponential backoff using increasing amounts of pause".